htdocs/core/class/commonobject.class.php

```diff
@@ -5213,14 +5213,10 @@ public function printOriginLine($line, $var, $restrictlist = '', $defaulttpldir
 			$discount->fk_soc = $this->socid;
 			$this->tpl['label'] .= $discount->getNomUrl(0, 'discount');
 		} elseif (!empty($line->fk_product)) {
-			$productstatic = new Product($this->db);
-			$productstatic->id = $line->fk_product;
-			$productstatic->ref = $line->ref;
-			$productstatic->type = $line->fk_product_type;
-			if (empty($productstatic->ref)) {
+			if (empty($line->product)) {
 				$line->fetch_product();
-				$productstatic = $line->product;
 			}
+			$productstatic = $line->product;
```

Comment:
With old code the $productstatic was always an object Product.

With new code, if the fetch fails $line->fk_product is not defined, the line->fetch_product will not load line->product and productstatic will be 0 or null making a fatal error later.
Should not hapen but we may protect with 
if (!is_object($productstatic)) {
   $productstatic = new Product($this->db);
}
after the
$productstatic = $line->product;